### PR TITLE
`Development`: Improve client test coverage for authentication

### PR DESCRIPTION
**Pull request by willergeorg**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).

### Motivation and Context

This PR improves the test coverage of folder `app/core/auth`
### Steps for Testing

Code review only

#### Code Review
- [x] Review 1
- [x] Review 2

### Test Coverage

| File name | Branch coverage (%) | Line coverage (%) |
| -- | -- | -- |
| `app/core/auth` | 85.04 | 90.11 |
| `account.service.ts` | 86.92 | 86.66 |
| `auth-jwt.service.ts` | 83.33 | 100 |
| `state-storage.service.ts` | 75 | 100 |
| `user-route-acess.service.ts` | 79.48 | 88.09 |

### CODE REVIEW

**Review by b-fein — COMMENTED** [review]
Tests look good.
I have two small questions where I’m not sure if the assertion can actually fail due to how `async` works.

File: src/test/javascript/spec/service/user-route-access.service.spec.ts [hunk 1]
```diff
@@ -62,4 +76,21 @@ describe('UserRouteAccessService', () => {
         service.canActivate(snapshot, routeStateMock);
         expect(MockSyncStorage.retrieve('authenticationToken')).toEqual('testToken');
     });
+
+    it('should return true if authorities are omitted', () => {
+        expect(service.checkLogin([], url)).resolves.toBe(true);
```
[the remark this answers is not in the document]
Comment (willergeorg) [reply src/test/javascript/spec/service/user-route-access.service.spec.ts]: Good catch, the await annotation is necessary apparently. Interesting is, that without it, if you change the expectation, the test will still succeed but an error message is shown that your promise does not resolve to the expected value.

File: src/test/javascript/spec/service/auth-jwt.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            expect(localStorageStoreSpy).not.toHaveBeenCalled();
+        });
+    });
+
+    describe('test authentication token removal', () => {
+        it('should clear only the authentication token', () => {
+            localStorageService.store(tokenKey, storedToken);
+            sessionStorageService.store(tokenKey, storedToken);
+
+            service.removeAuthTokenFromCaches().subscribe((resp) => {
```
[the remark this answers is not in the document]
Comment (willergeorg) [reply src/test/javascript/spec/service/auth-jwt.service.spec.ts:141]: Yes, thanks for pointing it out. I added the keywords and additionally upgraded every test case concerning a Promise with `async` and `await` keywords in 40d4ba2

**Review by b-fein — APPROVED** [review]
:+1:

**Review by cremertim — APPROVED** [review]
👍

**Review by canberkanar — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**